src/common/libflux/message.c

```diff
@@ -87,6 +87,14 @@ struct route_id {
     char id[0];                 /* variable length id stored at end of struct */
 };
 
+/* 'aux' for any auxiliary data user may wish to associate with iovec,
+ * user is responsible to free/destroy */
+struct msg_iovec {
+    const void *data;
+    size_t size;
+    void *aux;
```

Comment:
The name `aux` is slightly overloaded here.  We already have a `struct aux_item` in the message struct, as well as the aux `uint32_t` fields in the PROTO block.  Man I am having a hard time thinking of an alternative that is not "data" though.  Maybe `transport_data`?